src/V3Const.cpp

```diff
@@ -3694,9 +3694,9 @@ class ConstVisitor final : public VNVisitor {
     // win if bit select is a constant (otherwise we may need to compute bit index several times)
     TREEOPV("AstSel{$fromp.castBufIf1}",                "replaceSelIntoBiop(nodep)");
     TREEOPV("AstSel{$fromp.castNot}",                   "replaceSelIntoUniop(nodep)");
-    TREEOPV("AstSel{$fromp.castAnd,$lhsp.castConst}",   "replaceSelIntoUniop(nodep)");
-    TREEOPV("AstSel{$fromp.castOr,$lhsp.castConst}",    "replaceSelIntoUniop(nodep)");
-    TREEOPV("AstSel{$fromp.castXor,$lhsp.castConst}",   "replaceSelIntoUniop(nodep)");
+    TREEOPV("AstSel{$fromp.castAnd,$fromp.castConst}",   "replaceSelIntoUniop(nodep)");
+    TREEOPV("AstSel{$fromp.castOr,$fromp.castConst}",    "replaceSelIntoUniop(nodep)");
+    TREEOPV("AstSel{$fromp.castXor,$fromp.castConst}",   "replaceSelIntoUniop(nodep)");
```

Comment:
Was/is this a bug here? lhsp and fromp are the same node, cant be both types.

Reply:
These are looking for a select where the bit to select from is a constant.  Please make a pre-pull to fix this as it's an algorithm change, feel free to merge if it passes.  Likewise please prepull (together or separate) any other algorithm changes because of this good cleanup.

Reply:
Ok, what I have here still has the bug so this is not a fix for the algorithm. I will do a new pull after to fix that.